go/libraries/doltcore/sqle/dfunctions/dolt_checkout.go

```diff
@@ -0,0 +1,219 @@
+// Copyright 2021 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dfunctions
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/dolthub/go-mysql-server/sql"
+	"github.com/dolthub/go-mysql-server/sql/expression"
+
+	"github.com/dolthub/dolt/go/cmd/dolt/cli"
+	"github.com/dolthub/dolt/go/cmd/dolt/errhand"
+	"github.com/dolthub/dolt/go/libraries/doltcore/doltdb"
+	"github.com/dolthub/dolt/go/libraries/doltcore/env"
+	"github.com/dolthub/dolt/go/libraries/doltcore/env/actions"
+	"github.com/dolthub/dolt/go/libraries/doltcore/sqle"
+)
+
+const DoltCheckoutFuncName = "dolt_checkout"
+
+type DoltCheckoutFunc struct {
+	expression.NaryExpression
+}
+
+func (d DoltCheckoutFunc) Eval(ctx *sql.Context, row sql.Row) (interface{}, error) {
+	dbName := ctx.GetCurrentDatabase()
+
+	if len(dbName) == 0 {
+		return 1, fmt.Errorf("Empty database name.")
+	}
+
+	dSess := sqle.DSessFromSess(ctx.Session)
+	dbData, ok := dSess.GetDbData(dbName)
+
+	if !ok {
+		return 1, fmt.Errorf("Could not load database %s", dbName)
+	}
+
+	ap := cli.CreateCheckoutArgParser()
+	args, err := getDoltArgs(ctx, row, d.Children())
+
+	if err != nil {
+		return 1, err
+	}
+
+	apr := cli.ParseArgs(ap, args, nil)
+
+	if (apr.Contains(cli.CheckoutCoBranch) && apr.NArg() > 1) || (!apr.Contains(cli.CheckoutCoBranch) && apr.NArg() == 0) {
+		return 1, errors.New("Improper usage.")
+	}
+
+	// Checking out new branch.
+	if newBranch, newBranchOk := apr.GetValue(cli.CheckoutCoBranch); newBranchOk {
+		if len(newBranch) == 0 {
+			err = errors.New("error: cannot checkout empty string")
+		} else {
+			err = checkoutNewBranch(ctx, dbData, newBranch, "")
+		}
+
+		if err != nil {
+			return 1, err
+		}
+
+		return 0, nil
+	}
+
+	name := apr.Arg(0)
+
+	if len(name) == 0 {
```

Comment:
Seems like this should get caught by one of the method called below, no?

Reply:
In the normal checkout it's explicitly handled like this. I can handle this in the other functions as well.